packages/web/lib/browser.js

```diff
@@ -37,17 +68,15 @@ const getBrowserScope = (parent) => ({
         key,
         functions: {
           page:
-            "Defines a page definition which matches on the regex passed in as the first argument, and which executes the rule set passed in as the second argument on every matching page",
-          visit: "Open a new tab/page with the given url",
+            "Instantiates a page definition which matches on the regex passed in as the first argument, and which executes the rule set passed in as the second argument on every matching page",
+          visit:
+            "Open a new tab/page with the given url and checks for matches on that URL. If there are matches the correspond ruleSets will be run. If there is no match, a new page will be opened in the browser scope",
         },
       });
       return null;
     },
-    page: (scope) => (pattern, ...rulesets) => {
-      if (rulesets.length > 1) {
-        console.warn("Only one ruleset is currently supported");
-      }
-
+    page: (scope) => (pattern, ...ruleSets) => {
+      validateScope((scope) => scope.internal.isBrowser, scope, true);
```

Comment:
once you make the change to `validateScope`, you'll need to wrap this in an `if` check, and if it's false, then throw a descriptive `BrowseError`